### ISSUE TITLE
stability: segfault in raft.appendEntry

### ISSUE THREAD

**Issue by mberhault** [cockroachdb/cockroach#10556]
node died on `blue` (`13.68.117.221`). This was noticed due to 0 QPS alerts. The node was restarted 2 hours later by the chaos monkey.

CC: @cockroachdb/stability 

build sha: 3e0f63d

panic:
```
I161108 20:43:48.194532 124 storage/replica_proposal.go:328  [n7,s7,r4230/40:/Table/51/1/588{34372…-56923…}] new range lease replica {7 7 40} 2016-11-08 20:42:59.920762537 +0000 UTC 57.238246697s following replica {8 8 44} 2016-11-08 20:41:29.759855244 +0000 UTC 1m30.160907293s [physicalTime=2016-11-08 20:43:48.194482718 +0000 UTC]
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x103634c]

goroutine 125 [running]:
panic(0x172d440, 0xc420018040)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc42000c120)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:185 +0x6e
panic(0x172d440, 0xc420018040)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/coreos/etcd/raft.(*raft).appendEntry(0xc428c4f1d0, 0xc439619cc0, 0x1, 0x1)
        /go/src/github.com/coreos/etcd/raft/raft.go:521 +0xdc
github.com/coreos/etcd/raft.(*raft).becomeLeader(0xc428c4f1d0)
        /go/src/github.com/coreos/etcd/raft/raft.go:621 +0x1ce
github.com/coreos/etcd/raft.stepCandidate(0xc428c4f1d0, 0x6, 0x1d, 0x1e, 0xda4, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/coreos/etcd/raft/raft.go:1013 +0x5a9
github.com/coreos/etcd/raft.(*raft).Step(0xc428c4f1d0, 0x6, 0x1d, 0x1e, 0xda4, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/coreos/etcd/raft/raft.go:779 +0x11ef
github.com/coreos/etcd/raft.(*RawNode).Step(0xc42465e480, 0x6, 0x1d, 0x1e, 0xda4, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/coreos/etcd/raft/rawnode.go:195 +0xc9
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processRaftRequest.func4(0xc42465e480, 0x7f957a3e0078, 0xc420018258, 0x7f957a462880)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:2925 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).withRaftGroupLocked(0xc420d86380, 0x1a49400, 0xc43d69f828, 0xacb6db, 0xc420d86458)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:492 +0x139
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).withRaftGroup(0xc420d86380, 0xc43d69f828, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:509 +0x92
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processRaftRequest(0xc42072ee00, 0x7f957a462880, 0xc42f8bf080, 0xc4204bcf30, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:2926 +0xb72
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processRequestQueue(0xc42072ee00, 0x1178)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3153 +0x180
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc42032a1e0, 0xc42000c120)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:241 +0x267
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:181 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc42000c120, 0xc4206c2800)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66
```

full log:
[cockroach.stderr.gz](https://github.com/cockroachdb/cockroach/files/579556/cockroach.stderr.gz)

**Comment by tamird** [comment]
Might be fixed on master? I merged Ben's PR a few hours ago that fixed
something similar.

**Comment by mberhault** [comment]
it could be, but we haven't seen this one very much, so better to check than hope it doesn't show up again.

**Comment by bdarnell** [comment]
Yeah, it's the same stack trace so I'm 99% sure this is fixed by #10530.

**Comment by mberhault** [comment]
ok, closing. We'll be upgrading blue (and all others) today.
